linux/LinuxProcessList.c

```diff
@@ -1640,8 +1684,13 @@ static bool LinuxProcessList_recurseProcTree(LinuxProcessList* this, openat_arg_
       }
       #endif
 
-      if (ss->flags & PROCESS_FLAG_LINUX_CGROUP) {
-         LinuxProcessList_readCGroupFile(lp, procFd);
+      /* The process' cgroup name is required for checking whether it runs in a container */
+      LinuxProcessList_readCGroupFile(lp, procFd);
+      char *cgroup = lp->cgroup;
```

Comment:
This causes unnecessary increase of the (already huge) local stack frame. No need to have this copied as it's not modified. If really necessary this should have been `const` anyway. Also try to reduce scope of variables as much as possible.